shared/business-rules/discounts/free-medium-pizza.ts

```diff
@@ -10,6 +10,6 @@ export const freeMediumPizza = {
         }
 
         let pizzas = sortByPrice(flatten(getBasePrices(basket.items.filter(x => x.tags.indexOf('pizza') !== -1 && x.version === 'Medium'))));
-        return pizzas[0].price;
+        return pizzas[0] ? pizzas[0].price : 0;
```

Comment:
This looks strange. I would rather put `pizzas.length > 0 ? pizzas[0].price : 0`